bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,317 @@
+          description: "file object to update"
+          required: true
+          schema:
+            $ref: "#/definitions/KfConfigFile"
+      responses:
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+  /projects/{project}/deployments:
+    post:
+      tags:
+        - "deployment"
+      summary: "Create a new kubeflow deployment on GCP"
+      security:
+        - BearerToken: []
+      operationId: "createDeployment"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "body"
+          name: "body"
+          description: "Deployment name"
+          required: true
+          schema:
+            $ref: "#/definitions/KfDeployment"
+      responses:
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+  /projects/{project}/deployments/{deployment}:
+    get:
+      tags:
+        - "deployment"
+      description: "Get kubeflow deployment status"
+      security:
+        - BearerToken: []
+      operationId: "getDeployment"
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "path"
+          name: "project"
+          description: "gcp project string id"
+          required: true
+          type: "string"
+        - in: "path"
+          name: "deployment"
+          description: "Deployment name"
+          required: true
+          type: "string"
+      responses:
+        200:
+          description: "deployment status"
+          schema:
+            $ref: "#/definitions/KfDeployment"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorMessage"
+definitions:
+  BasicAuthIngress:
+    type: "object"
+    properties:
+      username:
+        type: "string"
+      passwordHash:
+        type: "string"
+  DeploymentMetadata:
+    type: "object"
+    properties:
+      name:
+        type: "string"
+      namespace:
+        type: "string"
+      version:
+        type: "string"
+      email:
+        type: "string"
+      zone:
+        type: "string"
+  KfDeployment:
+    type: "object"
+    properties:
+      metadata:
+        $ref: "#/definitions/DeploymentMetadata"
+      sAClientId:
+        type: "string"
+      status:
+        type: "string"
+  ErrorMessage:
+    type: "object"
+    properties:
+      errorMessage:
+        type: "string"
+  IapIngress:
+    type: "object"
+    properties:
+      clientId:
+        type: "string"
+      clientSecret:
+        type: "string"
+  KfConfig:
+    type: "object"
+    properties:
+      metadata:
+        $ref: "#/definitions/DeploymentMetadata"
+      extraComponents:
```

Comment:
Should KfConfig have a spec and status and follow the K8s resource model?

Status could provide information about things like version; e.g. if the underlying source has changed this could be reported via status.

Also it allows us to think of the resource as a controller where spec describes desired state and status provides actual state.